onnxruntime/core/framework/session_state.h

```diff
@@ -59,6 +60,93 @@ struct MemoryPatternGroup;
 class MemoryInfo;
 #endif
 
+struct PartialGraphExecutionManager {
+  PartialGraphExecutionManager() {
+    graph_runs_counter_ = 0;
+  }
+
+  int64_t GetNewPartialGraphRunId() {
+    std::lock_guard<OrtMutex> lock(lock_);
+    int64_t new_run_id;
+    if (available_run_ids_.size() > 0) {
+      new_run_id = available_run_ids_.back();
+      available_run_ids_.pop_back();
+    } else {
+      new_run_id = graph_runs_counter_++;
+    }
+
+    ORT_ENFORCE(to_be_executed_runs_.find(new_run_id) == to_be_executed_runs_.end());
+
+    to_be_executed_runs_.insert(new_run_id);
+    return new_run_id;
+  }
+
+  bool IsValidRunId(int64_t run_id) {
+    std::lock_guard<OrtMutex> lock(lock_);
+    auto it_in_progress_runs = graph_runs_.find(run_id);
```

Comment:
>it_in_progress_runs = graph_runs_.find(run_id); [](start = 9, length = 47)

for better readability

bool is_inprogress_run = graph_runs_.find(run_id) != graph_runs_.end(); 
bool is_to_be_executed_runs = ...

return is_inprogress_run ^ is_to_be_executed_runs;